docs/code-review.md

```diff
@@ -0,0 +1,363 @@
+            resource:    nil,
+        },
+        {
+            name:        "when resource is not valid then error",
+            resource:    MalformedResource{},
+        },
+        {
+            name:        "when resource is valid then no error",
+            resource:    CorrectResource{},
+        },
+    }
+
+    for _, tt := range tests {
+        t.Run(tt.name, func(t *testing.T) {
+            t.Setenv("ENV_REQUIRED_FOR_TEST", "testValue")
+            testDir := t.TempDir()
+            err := functionName(testDir)
+            assert.NoError(t, err)
+        })
+    }
+}
+
+func Test_FunctionNameWithError(t *testing.T) {
+    t.Setenv("ENV_REQUIRED_FOR_TEST", "testValue")
+    testDir := t.TempDir()
+    err := functionName(MalformedResource{})
+    assert.NoError(t, err)
+}
+```
+
+If the test has many use cases, it will be grouped into several tests per result, for example:
+
+```golang
+func Test_FunctionNameThenErr(t *testing.T) {
+    var tests = []struct {
+        name         string
+        resource     Resource
+        expectedErr  bool
+    }{
+        {
+            name:        "when resource is nil then error",
+            resource:    nil,
+        },
+        {
+            name:        "when resource is not valid then error",
+            resource:    MalformedResource{},
+        },
+        {
+            name:        "when resource is not valid because of X",
+            resource:    MalformedResource{},
+        },
+        {
+            name:        "when resource fails because of proxy",
+            resource:    MalformedResource{},
+        },
+    }
+
+    for _, tt := range tests {
+        t.Run(tt.name, func(t *testing.T) {
+            t.Setenv("ENV_REQUIRED_FOR_TEST", "testValue")
+            testDir := t.TempDir()
+            err := functionName(testDir)
+            if tt.expectedErr {
+                assert.Error(t, err)
+            } else {
+                assert.NoError(t, err)
+            }
+        })
+    }
+}
+func Test_FunctionNameThenNoErr(t *testing.T) {
+    var tests = []struct {
+        name         string
+        resource     Resource
+    }{
+        {
+            name:        "when resource is type 1",
+            resource:    CorrectResource1{},
+        },
+        {
+            name:        "when resource is type 1 and has x feature",
+            resource:    CorrectResource1{
+                FeatureX: true,
+            },
+        },
+        {
+            name:        "when resource is type 2",
+            resource:    CorrectResource2{},
+        },
+    }
+
+    for _, tt := range tests {
+        t.Run(tt.name, func(t *testing.T) {
+            t.Setenv("ENV_REQUIRED_FOR_TEST", "testValue")
+            testDir := t.TempDir()
+            err := functionName(testDir)
+            if tt.expectedErr {
+                assert.Error(t, err)
+            } else {
+                assert.NoError(t, err)
+            }
+        })
+    }
+}
```

Comment:
Same comment as previous example, I think we need to review this example. Check in line 171 is not needed and can be replaced by just `assert.NoError(t, err)`